objectherkenning_openbare_ruimte/on_edge/detection_pipeline/components/data_detection.py

```diff
@@ -0,0 +1,345 @@
+import csv
+import logging
+import os
+import pathlib
+import shutil
+import time
+from typing import List
+
+import numpy as np
+import torch
+from cvtoolkit.helpers.file_helpers import delete_file
+from ultralytics import YOLO
+
+from objectherkenning_openbare_ruimte.on_edge.detection_pipeline.components.input_image import (
+    InputImage,
+)
+from objectherkenning_openbare_ruimte.on_edge.detection_pipeline.components.model_result import (
+    ModelResult,
+)
+from objectherkenning_openbare_ruimte.on_edge.utils import (
+    get_frame_metadata_csv_file_paths,
+    get_img_name_from_csv_row,
+    log_execution_time,
+    move_file,
+)
+from objectherkenning_openbare_ruimte.settings.settings import (
+    ObjectherkenningOpenbareRuimteSettings,
+)
+
+logger = logging.getLogger("detection_pipeline")
+
+
+class DataDetection:
+    def __init__(
+        self,
+    ):
+        """
+        Object that find containers in the images using a pre-trained YOLO model and blurs sensitive data.
+        """
+        settings = ObjectherkenningOpenbareRuimteSettings.get_settings()
+
+        self.images_folder = pathlib.Path(settings["detection_pipeline"]["images_path"])
+        self.detections_folder = settings["detection_pipeline"]["detections_path"]
+
+        self.training_mode = settings["detection_pipeline"]["training_mode"]
+        self.training_mode_destination_path = pathlib.Path(
+            settings["detection_pipeline"]["training_mode_destination_path"]
+        )
+
+        self.defisheye_flag = settings["detection_pipeline"]["defisheye_flag"]
+        self.defisheye_params = settings["detection_pipeline"]["defisheye_params"]
+
+        self._check_model_availability(settings["detection_pipeline"]["sleep_time"])
```

Comment:
Move this to below line 69, now you're using `self.pretrained_model_path` before it is declared.

Reply:
I modified the _check_model_availability, can you check it? I think it's more clear.

Reply:
Looks good, but you should still move the method call to after declaring self.pretrained_model_path (now on line 66).